catalog/admin/module/catalog-admin-module-maplayers/src/main/webapp/map-layers/index.js

```diff
@@ -237,7 +243,49 @@ let FixedHeader = ({ muiTheme: { palette }, disabled = true, onSave, onReset })
 
 FixedHeader = muiThemeable()(FixedHeader)
 
-const MapLayers = ({ onFetch, onUpdate, onSave, onReset, onMessage, disabled, providers = [], errors, loading, message }) => (
+const FixConfig = ({ buffer, error, onUpdate, onDiscard, onSave }) => {
+  const actions = [
+    <FlatButton
+      label='discard'
+      secondary
+      onClick={onDiscard}
+    />,
+    <RaisedButton
+      label='keep'
+      primary
+      disabled={error !== undefined}
+      keyboardFocused
+      onClick={onSave}
+    />
+  ]
+
+  return (
+    <Dialog title='Invalid Map Layer Configuration Found' actions={actions} open>
+      <Description>
+        Existing map layers configuration was found to be invalid. You can
+        edit the JSON here and keep or discard the existing configuration.
+      </Description>
+      <Error errorText={error}>
+        <AceEditor
+          mode='json'
+          theme='github'
+          fontSize={15}
+          tabSize={2}
+          width='100%'
+          height='400px'
+          editorProps={{
+            $blockScrolling: Infinity
+          }}
+          enableBasicAutocompletion
+          name='json-editor'
+          value={buffer}
+          onChange={onUpdate} />
+      </Error>
+    </Dialog>
+  )
+}
+
+const MapLayers = ({ onFetch, onUpdate, onSave, onReset, onMessage, onSetInvalid, onSet, disabled, providers = [], errors, loading, message, invalid }) => (
```

Comment:
`standard` doesn't have any configuration options. I didn't notice how long it was.